### PR TITLE
Added CALL ... AS OF

### PR DESCRIPTION
**Pull request by Hydrocharged**

### CODE REVIEW

**Review by fulghum — APPROVED** [review]
Very, very cool changes! This is such a neat feature to be able to control the version of the stored procedure definition, as well as the version of the data it operates on. 🙌

File: go/libraries/doltcore/sqle/procedures_table.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func DoltProceduresGetTable(ctx *sql.Context, db Database) (*WritableDoltTable,
 	}
 }
 
-func DoltProceduresGetAll(ctx *sql.Context, db Database) ([]sql.StoredProcedureDetails, error) {
+func DoltProceduresGetAll(ctx *sql.Context, db Database, procedureName string) ([]sql.StoredProcedureDetails, error) {
```
[the remark this answers is not in the document]
Comment (Hydrocharged) [reply go/libraries/doltcore/sqle/procedures_table.go:110]: Great idea!

File: go/libraries/doltcore/sqle/database.go [hunk 2]
```diff
@@ -1221,9 +1221,25 @@ func (db Database) DropTrigger(ctx *sql.Context, name string) error {
...
+// GetStoredProcedure implements sql.StoredProcedureDatabase.
+func (db Database) GetStoredProcedure(ctx *sql.Context, name string) (sql.StoredProcedureDetails, bool, error) {
+	//TODO: fix this to return a single procedure
+	name = strings.ToLower(name)
+	procedures, err := DoltProceduresGetAll(ctx, db, name)
+	if err != nil {
+		return sql.StoredProcedureDetails{}, false, nil
+	}
+	for _, procedure := range procedures {
+		if name == procedure.Name {
```
[the remark this answers is not in the document]
Comment (Hydrocharged) [reply go/libraries/doltcore/sqle/database.go:1233]: I meant to check if we lowercase procedure names when we insert them into the table. Completely forgot to do that, I'll fix it if we don't or add a note if we do.

**Comment by Hydrocharged** [comment]
Failed bats test is a flake, merging anyway

**Pull request merged**